internal/controller/controllers/cluster_controller_test.go

```diff
@@ -132,18 +142,132 @@ var _ = Describe("cluster reconcile", func() {
 			cluster := newCluster(clusterName, testNamespace, defaultClusterSpec)
 			Expect(c.Create(ctx, cluster)).ShouldNot(HaveOccurred())
 
+			request := newClusterRequest(cluster)
+			result, err := cr.Reconcile(request)
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(ctrl.Result{RequeueAfter: defaultRequeueAfterOnError}))
+
+			cluster = getTestCluster()
+			Expect(cluster.Status.Error).To(Equal(expectedError.Error()))
+		})
+	})
+
+	Context("cluster update", func() {
+		var (
+			sId     strfmt.UUID
+			cluster *v1alpha1.Cluster
+		)
+
+		BeforeEach(func() {
+			pullSecret := getDefaultTestPullSecret("pull-secret", testNamespace)
+			Expect(c.Create(ctx, pullSecret)).To(BeNil())
+
+			cluster = newCluster(clusterName, testNamespace, defaultClusterSpec)
+			id := uuid.New()
+			sId = strfmt.UUID(id.String())
+			cluster.Status = v1alpha1.ClusterStatus{
+				State: models.ClusterStatusPendingForInput,
+				ID:    id.String(),
+			}
+			Expect(c.Create(ctx, cluster)).ShouldNot(HaveOccurred())
+		})
+
+		It("update pull-secret network cidr and cluster name", func() {
+			backEndCluster := &common.Cluster{
+				Cluster: models.Cluster{
+					ID:                       &sId,
+					Name:                     "different-cluster-name",
+					OpenshiftVersion:         defaultClusterSpec.OpenshiftVersion,
+					ClusterNetworkCidr:       "11.129.0.0/14",
+					ClusterNetworkHostPrefix: defaultClusterSpec.ClusterNetworkHostPrefix,
+				},
+				PullSecret: "different-pull-secret",
+			}
+			mockInstallerInternal.EXPECT().GetClusterByKubeKey(gomock.Any()).Return(backEndCluster, nil)
+
+			updateReply := &common.Cluster{
+				Cluster: models.Cluster{
+					ID:         &sId,
+					Status:     swag.String(models.ClusterStatusInsufficient),
+					StatusInfo: swag.String(models.ClusterStatusInsufficient),
+				},
+				PullSecret: testPullSecretVal,
+			}
+			mockInstallerInternal.EXPECT().UpdateClusterInternal(gomock.Any(), gomock.Any()).Return(updateReply, nil)
+
 			request := newClusterRequest(cluster)
 			result, err := cr.Reconcile(request)
 			Expect(err).To(BeNil())
 			Expect(result).To(Equal(ctrl.Result{}))
 
-			key := types.NamespacedName{
-				Namespace: testNamespace,
-				Name:      clusterName,
+			cluster = getTestCluster()
+			Expect(cluster.Status.State).To(Equal(models.ClusterStatusInsufficient))
+			Expect(cluster.Status.StateInfo).To(Equal(models.ClusterStatusInsufficient))
```

Comment:
Should we also assert for `Cluster.Spec.PullSecret` ,  `Cluster.Spec.ClusterNetworkCidr` and `Cluster.Spec.Name` ?

Reply:
Spec is the one define those values and they will not change in the spec but in the DB, DB is mocked there so there is  no point to test it in unit tests, it's a good test case for subsystem for example, unless i didn't understand the comment.
What i can test here are the values that i set to `UpdateClusterInternal` maybe here it will be more releavnt

Reply:
Yup, my proposal is more applicable for subsystem tests.